### PR TITLE
Reject with error object on promise rejection in unlock

### PR DESCRIPTION
**Pull request by tmashuang**
Brought up in https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/103#issuecomment-914436533.

### CODE REVIEW

File: index.js
Comment (Gudahtt) [inline index.js:130]: Nit: We could consider this instead:

```suggestion
          reject(payload.error || new Error('Unknown error'))
```

If we trust that `payload.error` is an error (which it probably will be once #103 is merged). Right now it would double-wrap the error. But that's fine - an Error is still a valid constructor parameter for an Error. It just means the log will start with `Error: Error:` instead of `Error:`.

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: index.js
Comment (tmashuang) [reply index.js:130]: Should we do that with the rest of the file? I was following that pattern.

Comment (Gudahtt) [reply index.js:130]: Oh! Good point, I hadn't noticed this is how it was done elsewhere.

I guess we should change this everywhere, to avoid the duplicate `Error:` prefix, but it doesn't matter a great deal. We can merge this PR as-is if you want, since it does restore consistency.

Comment (tmashuang) [reply index.js:130]: I'll make another PR for it.

Comment (tmashuang) [reply index.js:130]: https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/105

**Pull request merged**
